packages/scripts/config/webpack.config.js

```diff
@@ -89,20 +83,38 @@ const config = {
 	},
 	module: {
 		rules: [
-			{
+			// Sourcemap Loader is a complement to the `devtool` configuration,
+			// and thus is only necessary for development environments.
+			! isProduction && {
 				test: /\.js$/,
 				use: require.resolve( 'source-map-loader' ),
 				enforce: 'pre',
 			},
 			{
 				test: /\.js$/,
 				exclude: /node_modules/,
-				use: {
-					loader: require.resolve( 'babel-loader' ),
-					options: getBabelLoaderOptions(),
-				},
+				use: [
+					'thread-loader',
+					{
+						loader: require.resolve( 'babel-loader' ),
+						options: {
+							// Babel uses a directory within local node_modules
+							// by default. Use the environment variable option
+							// to enable more persistent caching.
+							cacheDirectory: process.env.BABEL_CACHE_DIRECTORY || true,
+
+							// Provide a fallback configuration if there's not
+							// one explicitly available in the project.
+							...( ! hasBabelConfig() && {
```

Comment:
I had never liked the options weren't colocated with the loader, so I welcome this change!

Reply:
object destructuring magic :)